src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -911,6 +932,58 @@ struct data_retention
 
 };
 
+struct mailbox
+{
+    using result_type = boost::any;
+
+    static xcl_mailbox
+        init_mailbox_info(const xrt_core::device* dev)
+    {
+        xcl_mailbox info = { 0 };
+        userpf::get_mailbox_info(dev->get_user_handle(), &info);
+        return info;
+    }
+
+    static result_type
+        get_info(const xrt_core::device* device, key_type key)
+    {
+        static std::map<const xrt_core::device*, xcl_mailbox> info_map;
+        static std::mutex mutex;
+        std::lock_guard<std::mutex> lk(mutex);
+
+        auto it = info_map.find(device);
+        if (it == info_map.end()) {
+            auto ret = info_map.emplace(device, init_mailbox_info(device));
+            it = ret.first;
+        }
+
+        const xcl_mailbox& info = (*it).second;
+        int i;
+        std::vector<std::string> vec;
```

Comment:
scope this variable in the case where it is used.

Reply:
when tried to move vec variable declaration to the case where it is used, then seeing below compile time error is seen

```
  C:\Users\Administrator\Desktop\1july_xrt\XRT\src\runtime_src\core\pcie\windows\device_windows.cpp(762): error C2361:
initialization of 'vec' is skipped by 'default' label [C:\Users\Administrator\Desktop\1july_xrt\XRT\build\WRelease\runt
ime_src\core\pcie\windows\xrt_core.vcxproj]
```

Reply:
Did you scope the case with curly braces?